client/app/scripts/utils/web-api-utils.js

```diff
@@ -235,20 +241,19 @@ export function getResourceViewNodesSnapshot(state, dispatch) {
   getNodesForTopologies(state, dispatch, topologyIds);
 }
 
-// NOTE: getState is called every time to make sure the up-to-date state is used.
-export function getTopologies(getState, dispatch, initialPoll = false) {
+function pollTopologies(getState, dispatch, initialPoll = false) {
```

Comment:
Probably it wouldn't be that hard to put it all in one method especially with help of Promises, but my thinking was the following:

* There is very little redundancy right now in `getTopologiesOnce` as all the complex logic is behind `topologiesUrl`, `receiveTopologies` and `receiveError`.
* I wanted to follow the example of `getNodes` and to make an early split between frozen and polling state consistent across our API calls (maybe this pattern could be generalized somehow).
* I think the logical split between polling/websockets and one-time calls is good as it enables us to manage them independently - e.g. what if we wanted to make `pollTopologies` use websockets in the future?

Also, doing it like this was simpler for now, I am aware that big parts of this code should probably be redone from scratch :)